commit c60e4fdc0af8ae0474dc7b57f7445f478b0cd9fd
Author: Stephen Segal <ssegal@meraki.com>
Date:   Mon Feb 13 13:24:23 2017 -0800

    fix stdio writes to procfs
    
    When writing to /proc/sys using stdio, procfs sees the zero-length
    write in this writev call as a separate write, which causes the
    variable in question to be set to null.  So we should not include a
    zero-length write as part of the writev call.

diff --git a/src/stdio/__stdio_write.c b/src/stdio/__stdio_write.c
index d2d89475..da2e93e8 100644
--- a/src/stdio/__stdio_write.c
+++ b/src/stdio/__stdio_write.c
@@ -9,7 +9,8 @@ size_t __stdio_write(FILE *f, const unsigned char *buf, size_t len)
 	};
 	struct iovec *iov = iovs;
 	size_t rem = iov[0].iov_len + iov[1].iov_len;
-	int iovcnt = 2;
+	int tot_iovcnt = len > 0 ? 2 : 1;
+	int iovcnt = tot_iovcnt;
 	ssize_t cnt;
 	for (;;) {
 		cnt = syscall(SYS_writev, f->fd, iov, iovcnt);
@@ -21,7 +22,7 @@ size_t __stdio_write(FILE *f, const unsigned char *buf, size_t len)
 		if (cnt < 0) {
 			f->wpos = f->wbase = f->wend = 0;
 			f->flags |= F_ERR;
-			return iovcnt == 2 ? 0 : len-iov[0].iov_len;
+			return iovcnt == tot_iovcnt ? 0 : len-iov[0].iov_len;
 		}
 		rem -= cnt;
 		if (cnt > iov[0].iov_len) {
